examples/coappearances.cpp

```diff
@@ -0,0 +1,327 @@
+/**
+ * Copyright (c) 2018 HERE Europe B.V.
+ * See the LICENSE file in the root of this project for license details.
+ */
+
+#include "coappearances.hpp"  // generated by flatdata from coappearances.flatdata
+#include "picojson.h"
+
+#include <flatdata/flatdata.h>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+namespace co = coappearances;
+
+void
+convert_meta( const picojson::object& obj, co::GraphBuilder builder, std::string& strings )
+{
+    // Since flatdata's mutators are not holding any data, we are creating a vector with a single
+    // element for holding the data.
+    flatdata::Vector< co::Meta > data( 1 );
+    auto meta = data[ 0 ];
+    meta.title_ref = strings.size( );
+    strings += obj.at( "title" ).get< std::string >( ) + '\0';
+    meta.author_ref = strings.size( );
+    strings += obj.at( "author" ).get< std::string >( ) + '\0';
+    builder.set_meta( meta );
+}
+
+uint16_t
+convert_id( const std::string& id )
+{
+    assert( id.size( ) == 2 );
+    return ( static_cast< uint8_t >( id[ 0 ] ) << 8 ) + static_cast< uint8_t >( id[ 1 ] );
+}
+
+using CharactersIndex = std::map< uint16_t /* id */, uint16_t /* ref */ >;
```

Comment:
This might need a better comment as `id` on one side and `ref` on the other is a little bit confusing.

Reply:
Yes, makes sense. I also realized that I actually do not need `convert_id` function, since I am not storing ids anyway.